dfir_rs/src/util/slot_vec.rs

```diff
@@ -117,3 +127,59 @@ impl<Key: ?Sized, Val> Default for SlotVec<Key, Val> {
         Self::new()
     }
 }
+
+/// A secondary map used to associated data with keys from elements in an existing [`SlotVec`].
```

Comment:
where is this structure maintained consistent with the SlotVec?

Reply:
Only via the keys. Meant to use a different type `Tag`